src/Repositories/DatasetsRepository.php

```diff
@@ -132,23 +135,53 @@ public static function resolve(string $description, array $dataset): array|null
         return $namedData;
     }
 
+    /**
+     * @return Closure|iterable<int|string, mixed>
+     */
+    private static function getScopedDataset(string $name, string $currentTestFile)
+    {
+        $matchingDatasets = array_filter(self::$datasets, function (string $key) use ($name, $currentTestFile) {
+            [$datasetScope, $datasetName] = explode('>>>', $key);
```

Comment:
I don't really like that we carry this string around everywhere. Should we extract it to a constant?

Reply:
done